### PR TITLE
Issue 2141/loading state list

### LINKED ISSUE #2141: Lists: Missing loading state while creating a list from selection
## Description
It's possible to create a new list from the selection in another list. Doing so, however, can take a lot of time, and during that time there is no indication that anything is loading, which causes users to click again and again, which actually creates multiple lists.

## Steps to reproduce
1. Go to https://app.dev.zetkin.org/organize/1/people
2. Click on "Create"
3. Click on "Create list"
4. In the new list, configure Smart Search to select everyone in the organization
5. Select lots of people, e.g. all the people in the list using the checkbox on the left
6. Click the "Create list from selection" button

### Expected Behaviour
A new list should be created, with the selected people in it, and the user be navigated to the new list. While the list is being created, the button should be disabled and there should be some sort of loading state, e.g. a spinner inside the button.

### Actual Behaviour
A new list is created, with the selected people in it, after which the user is navigated to the new list. But while the list is being created, which can take several seconds or more, there is nothing indicating that this is happening and the button can still be clicked again.

## Screenshots (if you have any)
image

### PR DESCRIPTION
**Pull request by rebecarubio**
## Description
This PR adds a loading state when users create a new list from an old list.

## Screenshots

https://github.com/user-attachments/assets/1a8f64c2-2756-47a8-a2d0-e28b11ce4b13

## Changes
* Adds a `isLoading `variable to the state in the `useCreateView ` hook
* Adds a `CircularProgress` to be displayed when isLoading is true in `ViewDataTableToolbar `component

## Notes to reviewer
None

## Related issues
Resolves #2141

### CODE REVIEW

**Comment by ziggabyte** [comment]
Is there a particular reason you decided to do a loading state for the whole list, instead of just the button? (basing this on the suggested implementation in the issue, 

> some sort of loading state, e.g. a spinner inside the button.

**Comment by rebecarubio** [comment]
omg nooo, I'll change it to place the loader be inside the button! :)

File: src/features/views/hooks/useCreateView.ts [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+interface useCreateViewReturn {
+  createView: (folderId?: number, rows?: number[]) => void;
+  isLoading: boolean;
+}
+
+export default function useCreateView(orgId: number): useCreateViewReturn {
   const apiClient = useApiClient();
   const router = useRouter();
   const dispatch = useAppDispatch();
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (richardolsson) [inline src/features/views/hooks/useCreateView.ts:18]: Nice solution!

File: src/features/views/components/ViewDataTable/ViewDataTableToolbar.tsx [hunk 2]
```diff
@@ -55,16 +57,23 @@ const ViewDataTableToolbar: React.FunctionComponent<
...
-          startIcon={<Launch />}
-        >
-          <Msg id={messageIds.toolbar.createFromSelection} />
+      {!isLoading && (
+        <Slide direction="left" in={!!selection.length} timeout={150}>
+          <Button
+            data-testid="ViewDataTableToolbar-createFromSelection"
+            disabled={disabled}
+            onClick={onViewCreate}
+            startIcon={<Launch />}
```
Comment (richardolsson) [inline src/features/views/components/ViewDataTable/ViewDataTableToolbar.tsx:66]: How about something like this instead?

```suggestion
            disabled={disabled || isLoading}
            onClick={onViewCreate}
            startIcon={isLoading? <CircularProgress size={25}> : <Launch />}
```

File: src/features/views/components/ViewDataTable/ViewDataTableToolbar.tsx [hunk 3]
```diff
@@ -55,16 +57,23 @@ const ViewDataTableToolbar: React.FunctionComponent<
...
     <Box role="toolbar">
-      <Slide direction="left" in={!!selection.length} timeout={150}>
-        <Button
-          data-testid="ViewDataTableToolbar-createFromSelection"
-          disabled={disabled}
-          onClick={onViewCreate}
-          startIcon={<Launch />}
-        >
-          <Msg id={messageIds.toolbar.createFromSelection} />
+      {!isLoading && (
```
Comment (richardolsson) [inline src/features/views/components/ViewDataTable/ViewDataTableToolbar.tsx:60]: Right now, you technically have two separate buttons for the loading vs not loading state. In my head, I think of them as the same button, but in two different states. That doesn't mean that they can't be implemented separately, but I see no reason to, so I would just remove this condition (and the entire other block) and just pass conditional properties to the single button based on `isLoading`.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Nice work! I find the button changing so drastically a bit jarring. Would it be an option to just replace the icon with a loading indicator and turn the button disabled?

File: src/features/views/components/ViewDataTable/ViewDataTableToolbar.tsx (hunk 3 above)
Comment (rebecarubio) [reply src/features/views/components/ViewDataTable/ViewDataTableToolbar.tsx:60]: I totally agree with this solution and avoiding having 2 buttons, I'll do the changes asap :)

**Review by richardolsson — APPROVED** [review]
Great work adapting this. I have tried the preview build and reviewed the code and it all looks good to me now. I know at least one organizer who will be very happy about this.

**Pull request merged**